README.md

```diff
@@ -69,7 +67,172 @@ More about ten pin bowling here: http://en.wikipedia.org/wiki/Ten-pin_bowling
+[index of rollList use] (https://github.com/s-dousse/bowling-challenge_JS/blob/main/images/diagram2.png)
+
+```
+getFinalScore()
+calculate score frame by frame
+    10 frames (use index of rollList)
+        (i = 0)
+        if strike ? (checking first roll of the frame : rollList[i] = 10)
+          sum = 10pts (strike) + bonus1 (first roll next frame => rollList[i+1]) + bonus2 (second roll next frame => rollList[i+2])
+          i + 1 (there is no 2nd roll : move to next frame)
+        else
+          if spare? (checking both rolls of the frame : rollList[i] + rollList[i+1] == 10 )
+            sum = 10ps + bonus1 (first roll next frame => rollList[i+1])
+          else
+            sum = total of both rolls: rollList[i] + rollList[i+1]
+          end
+          i + 2 (skip second roll of the current frame + move to the next frame (1st roll))
+        end
+    end
+```
+
+not so sure what's happening here, having a nightmare to get place the return at the right line to get sum back... and not undefined or NaN
+
+also I tried to use a function from the Class within another function of the same Class and it said undefined function so I will have to look into this.
+
+I uninstalled ESLint, it was bloking prettier and as I am new to JS prettier is just a lifesaver and I don't feel confortable working without it yet..
+
+Hard time debugging this one, my loop for iterating over all the 10 frames wasn't working:
+I need to return the value of sum in the if statement after calculatin the score for each frame. but then I couldn't change the index of my rollList after the return. so I created another variable (frameI) that I could change frame before calculating the score and return it.
+after struggling a bit it look like I can use this.sum when declare in the constructor function => I don't need to return the sum to update it
+
+Yay all tests passing : let's see how I can refactor my big chunky method getTotalScore to have clear responsability for each method + break in shorter methods:
+
+- isAStrike(i) => check the first roll of a frame
+- isASpare(i) => check if the sum of both rolls of a frame equal to 10
+- calulateStrikeScore()
+- calculateSpareScore()
+- caculateBasicScore()
+- moveToNextFrame(str)
+
+We can call these functions as methods within the getTotalScore function, but we use the `this` keyword!
+I think this is how it works:
+so `this` will point to the object that is calling the method
+in tests we do `game.getTotalScore()` (before const game = new BowlingGame())
+so in getTotalScore() we do
+`this.isAStrike(i)` will be like `game.isAStrike(i)`?
+also `this` behaves differently with arrow functions, I haven't used them here , but that's just luck and the tests are still passing after refactoring
+
+also not sure that having `this.sum` as class property is the best? but then I can pass it around the different function without using a argument
+or this is bad practice ? or I should do the same for the `this.rollList[?]`?
```

Comment:
Great questioning here!
As you suspect, using `this.sum` everywhere is not best practice. This is because it makes the code easier to break. A developer who doesn't know your code (or even yourself in a few weeks weeks :) ) could expect `this.sum` to be reliable as the current sum of the rolls at all times, which is not guaranteed when you use it as an intermediate summing variables.

Passing arguments is a good thing, and makes functions more independent and easier to understand separately. I'll give some suggestions for refactors.